skrifa/src/outline/autohint/latin/mod.rs

```diff
@@ -22,18 +22,41 @@ fn derived_constant(units_per_em: i32, value: i32) -> i32 {
     value * units_per_em / 2048
 }
 
-pub(crate) fn hint_outline(outline: &mut Outline, metrics: &UnscaledStyleMetrics, scale: &Scale) {
+/// Captures adjusted horizontal scale and outer edge positions to be used
+/// for horizontal metrics adjustments.
+#[derive(Copy, Clone, PartialEq, Default, Debug)]
+pub(crate) struct HintedMetrics {
+    pub x_scale: i32,
+    pub left_opos: i32,
+    pub left_pos: i32,
+    pub right_opos: i32,
+    pub right_pos: i32,
+}
+
+/// Applies the complete hinting process to a latin outline.
+///
+/// See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/autofit/aflatin.c#L3554>
+pub(crate) fn hint_outline(
```

Comment:
I don't find the interpretation of no HintedMetrics (None) very clear, suggest an additional comment to say what that means.

Reply:
This returns `None` when we have fewer than two edges because there isn't enough data to adjust metrics. Think of empty (spacing) glyphs or degenerate glyphs where all x coords are (roughly) the same. Noted as such.